packages/applications-service-api/src/services/submission.service.js

```diff
@@ -1,105 +1,41 @@
-const db = require('../models');
-const ApiError = require('../error/apiError');
-const { sendSubmissionNotification } = require('../lib/notify');
-const { getNIApplication } = require('./application.ni.service');
-
-const PLACEHOLDER_SUBMISSION_ID = 0;
-
-const createSubmission = async (submission) => {
-	try {
-		const submissionId = submission.submissionId || PLACEHOLDER_SUBMISSION_ID;
-
-		const dbSubmission = await db.Submission.create({
-			name: submission.name,
-			email: submission.email,
-			interestedParty: submission.interestedParty,
-			iPReference: submission.ipReference,
-			deadline: submission.deadline,
-			submissionType: submission.submissionType,
-			sensitiveData: submission.sensitiveData,
-			lateSubmission: submission.lateSubmission,
-			caseReference: submission.caseReference,
-			dateSubmitted: submission.dateSubmitted,
-			submissionId: submissionId,
-			representation: submission.representation
-		});
-
-		if (submissionId === PLACEHOLDER_SUBMISSION_ID) {
-			await updateSubmission({
-				id: dbSubmission.dataValues.id,
-				submissionId: dbSubmission.dataValues.id
-			});
-
-			return {
-				...dbSubmission.dataValues,
-				submissionId: dbSubmission.dataValues.id
-			};
-		}
-
-		return dbSubmission.dataValues;
-	} catch (e) {
-		if (
-			e.name === 'SequelizeDatabaseError' &&
-			e.parent.code === 'ER_TRUNCATED_WRONG_VALUE_FOR_FIELD'
-		) {
-			throw ApiError.badRequest(e.message);
-		} else {
-			throw e;
-		}
+const uuid = require('uuid');
+const config = require('../lib/config');
+const { createNISubmission, completeNISubmission } = require('./submission.ni.service');
+const { publishDeadlineSubmission } = require('./backoffice.publish.service');
+const { generateRepresentationPDF, uploadToBlobStorage } = require('../utils/file');
+
+const createSubmission = async (submission) =>
+	isBackOfficeSubmission(submission.metadata.caseReference)
+		? createBackOfficeSubmission(submission)
+		: createNISubmission(submission);
+
+const completeSubmission = async (submissionId) => completeNISubmission(submissionId);
+
+const createBackOfficeSubmission = async (submission) => {
+	const { metadata } = submission;
+
+	if (!metadata.submissionId) metadata.submissionId = uuid.v4();
```

Comment:
We might want to check this is ok, as currently users get:
Your reference number
12569

If these need to be communicated over the phone a uuid might be awkward.

Appreciate it's not easy without a DB.

Reply:
Yeah that's fair, UUID in this case is a bit overkill. I've pushed up a different approach based on case reference and timestamp. That should give enough uniqueness considering the volume of submissions